Add new version of replication controller details page

floreks opened the pull request:
This is the initial version of rc details page. I've added info card component that can be used to display info for different resources.

What's left:
- ~~Documentation~~
- ~~Add more tests~~
- ~~Check and remove unused/deprecated code~~ (partially done, to be finished in next PR)
- ~~Migrate all directives to components (in this view)~~
- ~~Make endpoints common components~~

I'd also refactor replication controller events to common component. Other resources will also display related events and there is single event object with common properties.

![zrzut ekranu z 2016-04-29 11-48-02](https://cloud.githubusercontent.com/assets/2285385/14913018/5214b598-0e00-11e6-9a6f-1fe46caadae8.png)
![zrzut ekranu z 2016-04-29 11-48-27](https://cloud.githubusercontent.com/assets/2285385/14913023/5506bb8e-0e00-11e6-9433-38776140a551.png)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubernetes/dashboard/694)
<!-- Reviewable:end -->

floreks commented:
@bryk could you review?

floreks commented:
Some checks are gone. Any idea why?

bryk commented:
> Some checks are gone. Any idea why?

No idea. Maybe codecov is down? I'll check probably on Wed

bryk commented:
Hmm... On serve:prod there are no card titles. Fix this. 

---

Reviewed 5 of 43 files at r1.
Review status: 3 of 46 files reviewed at latest revision, 22 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWnYYA1Bmq7q9od39e:-KGWnYYA1Bmq7q9od39f:60499036) (no related file):*
Looks really nice! A few comments. Remember about the tests and docs :) 

---

*[src/app/frontend/_variables.scss, line 54 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWecYFEOcp7OwkFTSq:-KGWecYGqmBTXZ0SIADh:1740129636) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/_variables.scss#L54)):*
AFAIK there are places which have this font weight. Can you make sure they use this var? 

---

*[src/app/frontend/common/components/contentcard/contentcard.scss, line 28 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWgp0cwqribdBCff23:-KGWgp0cwqribdBCff24:286817663) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard.scss#L28)):*
Hmm... Why do we need this? 

---

*[src/app/frontend/common/components/contentcard/contentcard.scss, line 33 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWem4N8snv0kS9YjRG:-KGWem4N8snv0kS9YjRH:68296682) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard.scss#L33)):*
Use the variable? 

---

*[src/app/frontend/common/components/contentcard/contentcard_component.js, line 23 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWgvWUuEKES_ZEGE7c:-KGWgvWUuEKES_ZEGE7d:1138233643) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard_component.js#L23)):*
Add `@type` info to this private var. 

---

*[src/app/frontend/common/components/contentcard/contentcard_module.js, line 30 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWh28tmnpSDE71nOkW:-KGWh28tmnpSDE71nOkX:-1846858452) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard_module.js#L30)):*
Can you create a separate module for this and include it herer? It's better to have one module per folder, rather than put component in sibling or parent modules. A few lines, but it'll allow to easy move stuff around.  

---

*[src/app/frontend/common/components/contentcard/infocard/infocard.html, line 1 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWhPhy9F5B5_ParZ-E:-KGWhPhzj7tHU9Qws-rH:-1773151809) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocard.html#L1)):*
How about moving all info card code one folder up? It feels like a good top level component which does not need to be hidden here. The fact that is uses content card underneath does not make it less important. 

---

*[src/app/frontend/common/components/contentcard/infocard/infocard.scss, line 17 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWi--iiYF1zffNf7iD:-KGWi--j1P-SlVrJ-uN2:617424490) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocard.scss#L17)):*
Is this class name okay? It feels like it is a group of info cards not a group inside an info card. How about making it just info-card-content? 

---

*[src/app/frontend/common/components/contentcard/infocard/infocard_component.js, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWiAMWf3rT4uX39LO8:-KGWiAMWf3rT4uX39LO9:1801094918) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocard_component.js#L16)):*
Yeah, add :) 

---

*[src/app/frontend/common/components/contentcard/infocard/infocardentry.html, line 18 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWiHgw78sItO9BMJzy:-KGWiHgw78sItO9BMJzz:-1523901494) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocardentry.html#L18)):*
nbsp? Why do you need it? It is usually easy to avoid with styles and/or layout. 

---

*[src/app/frontend/common/components/contentcard/infocard/infocardentry.scss, line 20 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWifKo1wTEUC8omKal:-KGWifKp-Qbthq8M6Stc:-2130134376) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocardentry.scss#L20)):*
Just $baseline-grid? Remember, this is the basic unit of organization of CSS in the poroject. It can be divided by 2 if it is padding top and bottom. 8px look nice


Or, you can  fix the kd-labels to include padding on bottom when needed. 

---

*[src/app/frontend/common/components/contentcard/infocard/infocardentry.scss, line 25 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWjpPS4Cj9htzNndVD:-KGWjpPS4Cj9htzNndVE:-1544158519) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocardentry.scss#L25)):*
Hmm... I knew something was different :) This light font for normal text makes it harder to red. How about making the titles bold? And keep the content untouched. 

---

*[src/app/frontend/common/components/contentcard/infocard/infocardgroup.html, line 19 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkD5ngCazQYeHLkpx:-KGWkD5ngCazQYeHLkpy:-1371383826) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocardgroup.html#L19)):*
="entry"? Do you need this? 

---

*[src/app/frontend/common/components/middleellipsis/middleellipsis_directive.js, line 48 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkTwgraK0ZihFvMcI:-KGWkTwgraK0ZihFvMcJ:1880426966) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/middleellipsis/middleellipsis_directive.js#L48)):*
Why this change? 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.html, line 17 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWki5AIIvwINXxv_rM:-KGWki5AIIvwINXxv_rN:-1439151476) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.html#L17)):*
Is `flex` needed here? I think not. 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.scss, line 45 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkq2aBpYFroqD_9dk:-KGWkq2aBpYFroqD_9dl:1588727193) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.scss#L45)):*
Hmmm... There's still much CSS code here? Do you see any chance for removing it entirely? 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html, line 30 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWlh3_tXRzwp5Gubfb:-KGWlh3_tXRzwp5Gubfc:373345223) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html#L30)):*
with-statuses="false"? 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html, line 37 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWlEwbs4ReZjK2-Ard:-KGWlEwbs4ReZjK2-Are:-315154433) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html#L37)):*
Make it small and nogrow? 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.scss, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWlNj7W1z0Ht2Fnn5b:-KGWlNj81vkdWKVCGVC4:-1120190726) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.scss#L16)):*
How can we get rid of this style here?

No need for action. I'm exploring ideas.

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerpods.html, line 23 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWml5f2Iyqj_5HB_kr:-KGWml5f2Iyqj_5HB_ks:1309524390) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerpods.html#L23)):*
We could iterate on sizes and grow patterns here later. But I dont want to make this PR bigger .

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerpods_component.js, line 18 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWmsH_T6r_1l30dJ6M:-KGWmsH_T6r_1l30dJ6N:647814929) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerpods_component.js#L18)):*
I think this will evolve into pod list card that will be reused here. And you? 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerservices_component.js, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWnP0zIilKxOEqHMt0:-KGWnP0zIilKxOEqHMt1:-1317213751) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerservices_component.js#L15)):*
Same here. This can evolve into the true one service card, right? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-:-KGWgO7s3S0e6GPp1e5-:199884595)*
<!-- Sent from Reviewable.io -->

floreks commented:
I'm always forgetting about jsdoc and adding `export` annotations. I didn't even notice that on prod. Thanks.

---

Review status: 2 of 47 files reviewed at latest revision, 22 unresolved discussions.

---

*[src/app/frontend/_variables.scss, line 54 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWecYFEOcp7OwkFTSq:-KGWoHHkXx1DzTwI_GJW:284041571) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/_variables.scss#L54)):*
I'll check.

---

*[src/app/frontend/common/components/contentcard/contentcard.scss, line 28 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWgp0cwqribdBCff23:-KGWoh8UzJ7wQlXQLBl3:-254023592) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard.scss#L28)):*
This is border line between card title and card content. To separate it.

---

*[src/app/frontend/common/components/contentcard/contentcard.scss, line 33 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWem4N8snv0kS9YjRG:-KGX-KMWesjl37D___vW:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard.scss#L33)):*
Done.

---

*[src/app/frontend/common/components/contentcard/contentcard_component.js, line 23 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWgvWUuEKES_ZEGE7c:-KGX-MBuWReq9cMdrG7v:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard_component.js#L23)):*
Done.

---

*[src/app/frontend/common/components/infocard/infocard.scss, line 17 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWi--iiYF1zffNf7iD:-KGX-Qk9U6JsFSrtOyIV:-744669867) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/infocard/infocard.scss#L17)):*
I agree it doesn't feel right. Done.

---

*[src/app/frontend/common/components/middleellipsis/middleellipsis_directive.js, line 48 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkTwgraK0ZihFvMcI:-KGWrg-A829xfZbAaAvr:1060990836) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/middleellipsis/middleellipsis_directive.js#L48)):*
Message was no longer true. Direct parent is now taken and there is no search for element with this class. This css class is not used in the code at all.

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.html, line 17 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWki5AIIvwINXxv_rM:-KGWsJtjmxetC897BgjD:-22540039) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.html#L17)):*
Yeah. Leftover from testing.

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.scss, line 45 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkq2aBpYFroqD_9dk:-KGWsPm8DdW7ByhzZy6A:-921877062) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.scss#L45)):*
Maybe not entirely but mostly. I've not yet checked and cleaned everything. There is still some code to remove but this PR is already big. I'll clean up everything in next one. Is that ok?

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html, line 30 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWlh3_tXRzwp5Gubfb:-KGWsivaTtlU2nK1kw5E:-1690593687) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html#L30)):*
I've used it to display events warnings icon. Status seems to be perfect for this.

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html, line 37 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWlEwbs4ReZjK2-Ard:-KGWxCQ1BG1NpX8rHjm4:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html#L37)):*
Done.

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.scss, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWlNj7W1z0Ht2Fnn5b:-KGWtKaQHD9wKNV4lbuD:1520848695) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.scss#L16)):*
This is mostly already existing code that I've moved to new components.

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerpods_component.js, line 18 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWmsH_T6r_1l30dJ6M:-KGWtboISTU6ZNWGEVuy:-556637567) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerpods_component.js#L18)):*
hmm... it definitely has potential to be migrated into some common component. For now I've just refactored current code into smaller pieces so it will be easier to extract components later.

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerservices_component.js, line 15 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWnP0zIilKxOEqHMt0:-KGWu1lW0uYQ6wrRrIdS:-485110604) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerservices_component.js#L15)):*
Sure.

---

*[src/app/frontend/common/components/contentcard/contentcard_module.js, line 30 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWh28tmnpSDE71nOkW:-KGWovIuoHsLy8TTzb9O:-842062816) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard_module.js#L30)):*
Makes sense.

---

*[src/app/frontend/common/components/contentcard/infocard/infocard.html, line 1 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWhPhy9F5B5_ParZ-E:-KGWp-MfmXaHAZLELsJR:1614445418) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocard.html#L1)):*
That's actually why I've added it there. I'll move it up.

---

*[src/app/frontend/common/components/contentcard/infocard/infocardentry.html, line 18 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWiHgw78sItO9BMJzy:-KGWpq8n82-HemhfFYLG:-1947871128) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocardentry.html#L18)):*
Added left margin on entry content

---

*[src/app/frontend/common/components/contentcard/infocard/infocardentry.scss, line 20 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWifKo1wTEUC8omKal:-KGWqlsHyA8mS6q0_o2y:-1692233861) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocardentry.scss#L20)):*
This is actually space between every entry. Making it `$baseline-grid` will require increasing padding on `kd-info-card-group-name` to `2 * $baseline-grid` otherwise space between group title and entries would be the same. What do you think?

---

*[src/app/frontend/common/components/contentcard/infocard/infocardentry.scss, line 25 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWjpPS4Cj9htzNndVD:-KGWqBfTsRMRd8yxT9ef:-1937245420) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocardentry.scss#L25)):*
Yeah I was thinking about making it bold, but group title is already bold and then entry title would also be bold. They look equally important when both are bold. What do you think?

---

*[src/app/frontend/common/components/contentcard/infocard/infocardgroup.html, line 19 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkD5ngCazQYeHLkpx:-KGWrIq0BCksGg9KLSm9:873080619) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/infocard/infocardgroup.html#L19)):*
It's just to make sure that group will always have an entry. Without that it will throw an error. Something like require.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-:-KGWu5erunKls7r4O2Rj:1130503023)*
<!-- Sent from Reviewable.io -->

floreks commented:
Review status: 2 of 49 files reviewed at latest revision, 22 unresolved discussions.

---

*[src/app/frontend/_variables.scss, line 54 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWecYFEOcp7OwkFTSq:-KGX8blK4D0WAfYpY7Y2:-499146009) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/_variables.scss#L54)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

floreks commented:
PTAL

bgrant0607 commented:
See comments on #627. Many apply here.

Also, it would be nice to keep details in sync with kubectl describe.

Speaking of kubectl describe, what about events? A timeseries graph would be awesome.

Other related resources: HorizontalPodAutoscaler, Secret, ConfigMap?

cc @janetkuo @pwittrock

bgrant0607 commented:
I see events is a separate tab.

Buttons to get the yaml/json would be nice, too.

floreks commented:
> Also, it would be nice to keep details in sync with kubectl describe.

I'll check that. I believe they are in sync now. Only some entries key names (Label selector instead of selector) may differ.

> Speaking of kubectl describe, what about events? A timeseries graph would be awesome.

For now events are displayed in the same way as they were. Timeseries graph sounds great. We'll add it to our todo list.

> Other related resources: HorizontalPodAutoscaler, Secret, ConfigMap?

All related resources will be added in next iterations. For now only backend for services and pods is ready and I didn't wanted to make this PR even bigger.

> Buttons to get the yaml/json would be nice, too.

We've thought about that for pods but this is also valid for RC. I've 2 ideas for that.

1. Make simple export button with option to choose yaml/json format.
2. Make dialog with yaml/json preview and export button to download the file.

I'm not sure about use cases here. Second one seems better if someone just wants to get specific information from yaml and not the whole file.

What do you think?

bryk commented:
Reviewed 18 of 52 files at r1, 12 of 16 files at r2, 18 of 27 files at r3, 6 of 7 files at r4, 3 of 4 files at r5, 2 of 2 files at r6.
Review status: 46 of 49 files reviewed at latest revision, 13 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWnYYA1Bmq7q9od39e:-KGv72JqCl7qyKooiFzR:29263257) (no related file):*
Can you make incremental coverage 100%? I.e., test everything here? :) 

Mostly reviewed. A few unresolved dicusssions. Will resolve soon. 

---

*[src/app/frontend/common/components/infocard/infocard.scss, line 22 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv3XatYMX338ic8j7d:-KGv3XatYMX338ic8j7e:-1552881435) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocard.scss#L22)):*
We still have this element whose name says it is a group of info cards. How about renaming it? To say that this is a section or something. 

---

*[src/app/frontend/common/components/infocard/infocardentry.html, line 19 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv4D1M4ZcGRjs7rFn_:-KGv4D1M4ZcGRjs7rFna:1111748121) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocardentry.html#L19)):*
Flex=80? Why do we need this? Cannot we just have flex=auto. Flex=80 means that we'll waste 20% of space. If you want to have space for the ctrl.title you can make it flex={none, nogrow or noshrink} 

---

*[src/app/frontend/common/components/infocard/infocardgroup.html, line 19 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkD5ngCazQYeHLkpx:-KGv54tgTLiZubM63Ske:-758083876) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/infocard/infocardgroup.html#L19)):*
Ok, I understand. 

---

*[src/app/frontend/common/components/middleellipsis/middleellipsis_directive.js, line 48 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkTwgraK0ZihFvMcI:-KGv5HdpvDOegrEwnvDA:-1600205003) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/middleellipsis/middleellipsis_directive.js#L48)):*
Yeah, you're correct :) Thanks 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.scss, line 45 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWkq2aBpYFroqD_9dk:-KGv5WBQ1OgDkEghiqke:255255699) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerdetail.scss#L45)):*
Yeah, this is OK. I was just asking. 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html, line 30 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWlh3_tXRzwp5Gubfb:-KGv5hERxDT6A2aFBa_H:1950693041) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.html#L30)):*
Sounds good. I didn't notice this. 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.scss, line 16 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWlNj7W1z0Ht2Fnn5b:-KGv5qfQ-3HNhO-LX17r:-183135561) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerevents.scss#L16)):*
Ok, maybe later when implementing new events view, we'll extract this to a common thing. 

---

*[src/app/frontend/replicationcontrollerdetail/replicationcontrollerpods_component.js, line 18 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWmsH_T6r_1l30dJ6M:-KGv6lodYUmQgz-55n-_:-485693544) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/replicationcontrollerdetail/replicationcontrollerpods_component.js#L18)):*
Yes. Just keep this in mind. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

floreks commented:
Review status: 46 of 49 files reviewed at latest revision, 7 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWnYYA1Bmq7q9od39e:-KGv7hjf_lck7cIeZKGu:-1622334477) (no related file):*
This codecov update is very misleading. It's not so clear what should be covered as it was before. I'll try to cover everything. :)

---

*[src/app/frontend/common/components/infocard/infocard.scss, line 22 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv3XatYMX338ic8j7d:-KGv852onW8OoanFA7rj:1041051991) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocard.scss#L22)):*
Is `info-card-section` better?

---

*[src/app/frontend/common/components/infocard/infocardentry.html, line 19 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv4D1M4ZcGRjs7rFn_:-KGv8l0Y7O0kAE9oYwO5:-243258610) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocardentry.html#L19)):*
When it is not set middle ellipsis behaves weird. Once it expands, it doesn't want to shrink the text.

![Zrzut ekranu z 2016-05-04 11-44-40.png](https://files.reviewableusercontent.io/images/reviewable/2285385/Ubv70fK4lDzX1kTp1S_YW08h/Zrzut%20ekranu%20z%202016-05-04%2011-44-40.png)

With flex auto
![Zrzut ekranu z 2016-05-04 11-46-54.png](https://files.reviewableusercontent.io/images/reviewable/2285385/qD6pmDwjJJKN4psQOsPTCmgN/Zrzut%20ekranu%20z%202016-05-04%2011-46-54.png)

With fixed flex
![Zrzut ekranu z 2016-05-04 11-49-52.png](https://files.reviewableusercontent.io/images/reviewable/2285385/3_4RA5W9MbGLit1xqsiHTppz/Zrzut%20ekranu%20z%202016-05-04%2011-49-52.png)

I'll try to find some other way to make it work.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

bryk commented:
*[src/app/frontend/common/components/infocard/infocardentry.html, line 19 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv4D1M4ZcGRjs7rFn_:-KGvD8JQffslI6wtKrgX:-524574949) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocardentry.html#L19)):*
I see. I had the same behavior inside resource card tables. It is hard to fix, I know. I cannot remember the exact solution, but it was about setting width, max-width to 100% and display block for some parent element. Maybe there was overflow:hidden also for some parent element. Look at the code of cards framework. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

bryk commented:
Review status: 46 of 49 files reviewed at latest revision, 7 unresolved discussions.

---

*[src/app/frontend/common/components/infocard/infocardentry.scss, line 20 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWifKo1wTEUC8omKal:-KGvKOg3DVZ_Xo2umcpA:-2114913479) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/infocard/infocardentry.scss#L20)):*
LGTM. We can iterate later if needed. 

---

*[src/app/frontend/common/components/infocard/infocardentry.scss, line 25 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWjpPS4Cj9htzNndVD:-KGvKS_gXJroucKuex8z:-1341033003) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/infocard/infocardentry.scss#L25)):*
Now it looks good. The section title is larger so it can be just bold. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

bryk commented:
*[src/app/frontend/common/components/infocard/infocard.scss, line 22 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv3XatYMX338ic8j7d:-KGvKezThbj3O7n-OLVS:-480064190) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocard.scss#L22)):*
Yea. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

bryk commented:
*[src/app/frontend/common/components/contentcard/contentcard.scss, line 28 \[r1\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWgp0cwqribdBCff23:-KGvKsuCPQp03KXgGPrj:676956979) ([raw file](https://github.com/kubernetes/dashboard/blob/6860f539eb7bb804abf2957973adc35eb9c01111/src/app/frontend/common/components/contentcard/contentcard.scss#L28)):*
Ok. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

bryk commented:
Review status: 46 of 49 files reviewed at latest revision, 4 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWnYYA1Bmq7q9od39e:-KGvKwYLUdiu4oQHYpK6:-1614347474) (no related file):*
PTAL.

You can view a better coverage report locally. I is much better. Check it out. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

floreks commented:
Review status: 46 of 49 files reviewed at latest revision, 3 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWnYYA1Bmq7q9od39e:-KGvNeXSwUJzAjx3bXd7:-1848484040) (no related file):*
I'll try it out. Thanks.

---

*[src/app/frontend/common/components/infocard/infocardentry.html, line 19 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv4D1M4ZcGRjs7rFn_:-KGvNA5AZBO4LXRWmQqI:-1992861483) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocardentry.html#L19)):*
I'm working on that now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

floreks commented:
Review status: 34 of 50 files reviewed at latest revision, 3 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGWnYYA1Bmq7q9od39e:-KGvknpR0Mh0oJbByEvh:676965300) (no related file):*
PTAL

---

*[src/app/frontend/common/components/infocard/infocardentry.html, line 19 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv4D1M4ZcGRjs7rFn_:-KGvkM-gRzaL561Z-N0R:-261358667) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocardentry.html#L19)):*
I've spent already too much time on this single issue and did not found solution. Maybe I'll just add todo there and if you have some time then you can take a look at that later?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

bryk commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 34 of 50 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/frontend/common/components/infocard/infocardentry.html, line 19 \[r6\]](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-KGv4D1M4ZcGRjs7rFn_:-KGvpFfvVO0WCWNPbhtE:2046524021) ([raw file](https://github.com/kubernetes/dashboard/blob/2351dc1b4eff1f78f086c43d9f0793308488733b/src/app/frontend/common/components/infocard/infocardentry.html#L19)):*
OK, please create an issue, assign it to me and reference it here. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-:-KGvpQli0pcJ1Ti7Ox8t:1417088721)*
<!-- Sent from Reviewable.io -->

bryk commented:
Reviewed 1 of 55 files at r1, 11 of 30 files at r3, 16 of 17 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694)*
<!-- Sent from Reviewable.io -->

bryk commented:
@floreks Please add the TODO and merge. 

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/dashboard/694#-:-KGvrToXEibyHOxhIZnw:-1321042572)*
<!-- Sent from Reviewable.io -->
